ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/attestation/utils/RewardBasedAttestationSorterTest.java

```diff
@@ -0,0 +1,365 @@
+
+  @BeforeEach
+  void setUp() {
+    final Spec actualSpec = TestSpecFactory.createMinimalAltair();
+    when(specVersion.miscHelpers()).thenReturn(actualSpec.getGenesisSpec().miscHelpers());
+    when(specVersion.beaconStateAccessors()).thenReturn(beaconStateAccessors);
+    when(specVersion.getMilestone()).thenReturn(SpecMilestone.ALTAIR);
+
+    when(state.toVersionAltair()).thenReturn(Optional.of(state));
+    when(spec.atSlot(any())).thenReturn(specVersion);
+    when(spec.getCurrentEpoch(state)).thenReturn(CURRENT_EPOCH);
+
+    when(state.getSlot()).thenReturn(STATE_SLOT);
+
+    // Default empty participation
+    final List<SszByte> initialParticipationBytes =
+        IntStream.range(0, VALIDATOR_COUNT).mapToObj(i -> SszByte.of((byte) 0)).toList();
+    final SszList<SszByte> initialParticipationSszList =
+        PARTICIPATION_LIST_SCHEMA.createFromElements(initialParticipationBytes);
+
+    when(state.getCurrentEpochParticipation()).thenReturn(initialParticipationSszList);
+    when(state.getPreviousEpochParticipation()).thenReturn(initialParticipationSszList);
+
+    // Default base reward for any validator
+    when(beaconStateAccessors.getBaseReward(eq(state), anyInt())).thenReturn(BASE_REWARD_VALUE);
+
+    sorter = RewardBasedAttestationSorter.create(spec, state);
+  }
+
+  @Test
+  void create_shouldReturnNOOP() {
+    when(specVersion.getMilestone()).thenReturn(SpecMilestone.PHASE0);
+
+    assertThat(RewardBasedAttestationSorter.create(spec, state)).isSameAs(NOOP);
+  }
+
+  @Test
+  void noop_shouldJustApplyLimit() {
+    final List<PooledAttestationWithData> attestationList =
+        List.of(
+            mock(PooledAttestationWithData.class),
+            mock(PooledAttestationWithData.class),
+            mock(PooledAttestationWithData.class));
+
+    assertThat(
+            NOOP.sort(attestationList, 2).stream()
+                .map(PooledAttestationWithRewardInfo::getAttestation))
+        .containsExactly(attestationList.getFirst(), attestationList.get(1));
+  }
+
+  @Test
+  void sort_shouldReturnEmptyList_whenInputIsEmpty() {
+    final List<PooledAttestationWithData> attestations = List.of();
+    final List<PooledAttestationWithRewardInfo> sortedAttestations = sorter.sort(attestations, 10);
+    assertThat(sortedAttestations).isEmpty();
+  }
+
+  @Test
+  void sort_shouldApplyLimit() {
+    final PooledAttestationWithData att1 =
+        mockPooledAttestationWithData(
+            STATE_SLOT.minus(1), CURRENT_EPOCH, List.of(0), List.of(TIMELY_SOURCE_FLAG_INDEX));
+    final PooledAttestationWithData att2 =
+        mockPooledAttestationWithData(
+            STATE_SLOT.minus(2), CURRENT_EPOCH, List.of(1), List.of(TIMELY_TARGET_FLAG_INDEX));
+
+    final List<PooledAttestationWithData> attestations = List.of(att1, att2);
+    final List<RewardBasedAttestationSorter.PooledAttestationWithRewardInfo> sortedAttestations =
+        sorter.sort(attestations, 1);
+
+    assertThat(sortedAttestations).hasSize(1);
```

Comment:
i was thinking the selected item should be important or is it not really relevant because mocked attestation?

Reply:
hmmm ...it isn't actually important 'cose the test aims only to check that the limit is applied.

Reply:
the actual output is checked in the other tests